File: 2696-minimum-string-length-after-removing-substrings/2696-minimum-string-length-after-removing-substrings.py
class Solution:
    def minLength(self, s: str) -> int:
        stack = []
        for c in s:
            stack.append(c)
            if len(stack) >= 2:
                sub = stack[-2] + stack[-1]
                if sub in ["AB", "CD"]:
                    stack.pop()
                    stack.pop()
        return len(stack)

File: 2696-minimum-string-length-after-removing-substrings/2696-minimum-string-length-after-removing-substrings.js
/**
 * @param {string} s
 * @return {number}
 */
var minLength = function(s) {
    const stack = []
    for (const c of s) {
        stack.push(c)
        if (stack.length >= 2) {
            const substring = stack.at(-2) + stack.at(-1)
            if (substring === "AB" || substring === "CD") {
                stack.pop()
                stack.pop()
            }
        }
    }
    return stack.length
};